source/slang/slang.cpp

```diff
@@ -123,7 +123,13 @@ CompileRequest::CompileRequest(Session* session)
 }
 
 CompileRequest::~CompileRequest()
-{}
+{
+    // delete things that may reference IR objects first
+    targets = decltype(targets)();
+    translationUnits = decltype(translationUnits)();
+    entryPoints = decltype(entryPoints)();
+    types = decltype(types)();
```

Comment:
Hmm... is the answer to my question about `canonicalTypes` that all types are now scoped to a compile request?